docs: add findtxout RPC documentation (#1200)
getfloresta/Floresta: doc/rpc/findtxout.md

```

### Examples

```bash
# Look for output 0 of a transaction, providing its scriptPubKey hex
floresta-cli findtxout aa5f3068b53941915d82be382f2b35711305ec7d454a34ca69f8897510db7ab8 0 0014c664139327b98043febeab6434eba89bb196d1af

# Same lookup, but start scanning from block height 800000 to speed things up
floresta-cli findtxout aa5f3068b53941915d82be382f2b35711305ec7d454a34ca69f8897510db7ab8 0 0014c664139327b98043febeab6434eba89bb196d1af 800000
```

## Arguments

`txid` - (string, required) The transaction id that created the output.

`vout` - (numeric, required) The index of the output within that transaction.

`script` - (string, required) The scriptPubKey of the output, hex encoded. This is what gets
matched against the block filters, so it must be correct for the output to be found.

`height_hint` - (numeric, optional, default=0) The block height to start scanning from. Providing
a height close to where the output was created avoids scanning the whole chain and makes the
lookup much faster.

## Returns

### Ok Response

If the output is found, an object describing it is returned:

- `value` - (numeric) The output value in satoshis.

> - `script_pubkey` - (string) The output scriptPubKey, hex encoded.

Reviewer: I think it's better to document the `gettxout` return format here and mention that this is a variant of it, since in the future this RPC will only return the `gettxout` response.

If the output can't be found (it doesn't exist or has already been spent), an empty object `{}`

> is returned.

Reviewer: I think this would fit better in the Notes section.

## Usage

### Synopsis

```
floresta-cli findtxout <txid> <vout> <script> [<height_hint>]
```

### Examples

```bash
# Look for output 0 of a transaction, providing its scriptPubKey hex
floresta-cli findtxout aa5f3068b53941915d82be382f2b35711305ec7d454a34ca69f8897510db7ab8 0 0014c664139327b98043febeab6434eba89bb196d1af

# Same lookup, but start scanning from block height 800000 to speed things up
floresta-cli findtxout aa5f3068b53941915d82be382f2b35711305ec7d454a34ca69f8897510db7ab8 0 0014c664139327b98043febeab6434eba89bb196d1af 800000
```

## Arguments

`txid` - (string, required) The transaction id that created the output.

`vout` - (numeric, required) The index of the output within that transaction.

`script` - (string, required) The scriptPubKey of the output, hex encoded. This is what gets
matched against the block filters, so it must be correct for the output to be found.

`height_hint` - (numeric, optional, default=0) The block height to start scanning from. Providing
a height close to where the output was created avoids scanning the whole chain and makes the
lookup much faster.

## Returns

### Ok Response

> If the output is found, an object describing it is returned. This is a variant of the
> [`gettxout`](gettxout.md) response, currently returning just:

Reviewer: ```suggestion
If the output is found, it returns the [`gettxout`](gettxout.md) response. It may also return the following variant:
```

- `value` - (numeric) The output value in satoshis.
- `script_pubkey` - (string) The output scriptPubKey, hex encoded.

> In the future `findtxout` will return the same response shape as `gettxout` directly.

Reviewer: remove this text
